src/site_upload/update_metadata/update_metadata.py

```diff
@@ -0,0 +1,63 @@
+"""Lambda for updating metadata
+
+Note: since this lambda is used with an EventSource of a FIFO queue,
+this lambda will not be allowed to be concurrently executed. Instead,
+it will be executed once, and fed up to ten events from that queue
+to be run in batch. Once it terminates, it will be re-invoked if there
+are remaining events in the queue.
+"""
+
+import json
+import os
+
+import boto3
+import botocore
+
+from shared import decorators, functions
+
+s3_client = boto3.client("s3")
+
+
+def update_source(metadata, update):
+    for update_key, update_val in update.items():
+        if isinstance(update_val, dict):
+            metadata[update_key] = update_source(
+                metadata[update_key] if update_key in metadata.keys() else {}, update[update_key]
+            )
+        else:
+            if not (
+                update_val is None
+                and update_key in metadata.keys()
+                and metadata[update_key] is not None
+            ):
+                metadata[update_key] = update_val
+    return metadata
+
+
+def process_event_queue(records):
+    sources = {}
+    metadata = {}
+    for record in records:
+        message = json.loads(record["body"])
+        if not any(x == message["key"] for x in sources.keys()):
+            sources[message["key"]] = []
+        updates = json.loads(message["updates"])
+        sources[message["key"]].append(updates)
+
+    for key, updates in sources.items():
```

Comment:
So nothing else is gonna run while this is up, because of the FIFO queue stuff, yeah? i.e. we don't have to worry about two of this lambda ever running at the same time, and no other bits of code are writing to these metadata paths?

Reply:
Correct - this is a property of subscribing a lambda to a FIFO queue.